atlasdb-config/src/main/java/com/palantir/atlasdb/http/FailoverFeignTarget.java

```diff
@@ -150,10 +151,13 @@ private void checkAndHandleFailure(RetryableException ex) {
 
 
     private void pauseForBackoff(RetryableException ex) {
-        double pauseTime = Math.pow(
+        double exponentialPauseTime = Math.pow(
                 GOLDEN_RATIO,
                 numSwitches.get() * failuresBeforeSwitching + failuresSinceLastSwitch.get());
-        pauseForBackoff(ex, Math.round(pauseTime));
+        long cappedPauseTime = Math.min(maxBackoffMillis, Math.round(exponentialPauseTime));
+        long pauseTimeWithJitter = ThreadLocalRandom.current().nextLong(cappedPauseTime);
+
+        pauseForBackoff(ex, pauseTimeWithJitter);
     }
 
     @VisibleForTesting
```

Comment:
shouldn't we actually apply the jitter below so that we hit all backoffs?

```
void pauseForBackoff(RetryableException ex, long pauseTime) {
    long timeout = ThreadLocalRandom.current().nextLong(Math.min(maxBackoffMillis, pauseTime));
```